### ISSUE TITLE
Spec should fail if test case name is duplicated

### ISSUE THREAD

**Issue by jankotek** [kotlintest/kotlintest#319]
StringSpec (and probably others) does not fail if there are multiple test cases with the same name. In example bellow, second test case is ignored, only "first" is printed.

I think it should throw an exception when Spec is initialized and tests are discovered.
```

class Example : StringSpec() {
    init {
        "test case"{
            println("first")
        }
        "test case"{
            println("second")
        }
    }
}

**Comment by sksamuel** [comment]
This is fixed in master. If you want to try an early release, you can try 3.1.0-RC3.

**Comment by sksamuel** [comment]
Closing this ticket as confirmed fixed in the 3.1.0-RC3 and RC4.
